src/app.js

```diff
@@ -207,44 +207,36 @@ class App {
   }
 
   // Start the synchronization
-  startSync (mode, callback) {
+  startSync (mode) {
     this.config.saveMode(mode)
     log.info('Run first synchronisation...')
-    this.sync.start(mode, err => {
-      this.sync.stop(function () {})
-      if (err) {
-        log.error(err)
-        if (err.stack) { log.error(err.stack) }
-      }
-      __guardFunc__(callback, f => f(err))
-    })
+    return this.sync.start(mode)
   }
 
   // Stop the synchronisation
-  stopSync (callback) {
-    if (callback == null) { callback = function () {} }
-    if (this.sync) {
-      this.sync.stop(callback)
-    } else {
-      callback()
+  stopSync () {
+    if (!this.sync) {
+      return Promise.resolve()
     }
+    return this.sync.stop()
   }
 
   // Start database sync process and setup file change watcher
-  synchronize (mode, callback) {
-    if (this.config.isValid()) {
-      this.instanciate()
-      this.startSync(mode, callback)
-    } else {
+  synchronize (mode) {
+    if (!this.config.isValid()) {
       log.error('No configuration found, please run add-remote-cozy' +
                 'command before running a synchronization.')
-      __guardFunc__(callback, f => f(new Error('No config')))
+      throw new Error('No client configured')
```

Comment:
Maybe extract `isValid()` call + log/error throwing to some `Config#ensureValid()` method? wdyt?